### ISSUE TITLE
[MU4 Issue] Crash if text object is selected and you click the Appearance or More button

### ISSUE THREAD

**Issue by rgreen5** [musescore/MuseScore#12820]
**Describe the bug**

This definately affects fingering but I can also reproduce on guitar barre lines, which suggests that it may affect various text objects. The bug doesn't always manifest first time but if you keep repeating the steps below the program will probably freeze and crash eventually.

**To Reproduce**

1. Open the attached MS3 score. 
[fingering_crash.zip](https://github.com/musescore/MuseScore/files/9315910/fingering_crash.zip)
2. All the default panels should be present, and Properties should be in play.
3. Click on any fingering then click on "Appearance" in the Properties panel.
RESULT: You may get a hang then a crash. If this doesn't work first time, click off the fingering, select another one and repeat.

**Expected behavior**

The Appearance / More sections should open seamlessly.

**Platform information**

OS: Linux Mint 20.1, Arch.: x86_64, MuseScore version (64-bit): 4.0.0-2844416344, revision: github-musescore-musescore-ae252cd

**Other info**

See #12797?

**Comment by oktophonie** [comment]
I've just encountered something that may be similar.
This happens in the latest nightly, 16 August 2022. It doesn't seem to happen in the Alpha 2. I've observed this on Windows and Linux.

- Open any file (the basic 'Treble Clef' template will do)
- Select any item (the initial clef will do)
- Open the 'Appearance' popup from Properties
- Deselect the item / press Escape
- Select _the same item_ again
- Open the 'Appearance' popup again; a crash ensues

It seems to be the _second_ opening of 'Appearance' on any given item that triggers the crash.
